plugins/inputs/varnish/varnish.go

```diff
@@ -115,13 +149,35 @@ func (s *Varnish) Gather(acc telegraf.Accumulator) error {
 		}
 	}
 
-	out, err := s.run(s.Binary, s.UseSudo, s.InstanceName, s.Timeout)
+	admout, err := s.admrun(s.AdmBinary, s.UseSudo, s.InstanceName, s.Timeout)
+	if err != nil {
+		return fmt.Errorf("error gathering vcl list from varnishadm: %s", err)
+	}
+
+	admscanner := bufio.NewScanner(admout)
+	vclactive := ""
+	for admscanner.Scan() {
+		vcllist := strings.Fields(admscanner.Text())
+		if len(vcllist) == 0 {
+			continue
+		}
+
+		status := vcllist[0]
+		vclname := vcllist[3]
```

Comment:
For me the name would be in another one down `vclname := vcllist[4]`. What version are you using?

Reply:
good catch, in the moment of PR, vclname lived in position 3, now with last varnish the position change it. This is exit from varnishadm vcl.list between different versions:
varnish 6.0:
```active      auto/warm          0 boot```
varnish 6.6:
```active   auto    warm         0    boot```

Maybe make more sense execute ```varnishadm vcl.list -j```, this command give same between versions. But this option only are supported with varnish >= 6.

Reply:
That makes sense. It looks like [all versions prior to 6.0.7 unsupported](https://varnish-cache.org/releases/index.html) by varnish-cache now anyway